File: common/src/main/java/com/cyanspring/common/transport/IServerUserSocketService.java
package com.cyanspring.common.transport;

import java.util.List;

import com.cyanspring.common.IPlugin;

public interface IServerUserSocketService extends IPlugin {
	IUserSocketContext getContext(String key);
	List<IUserSocketContext> getContextByUser(String user);
	void setUserContext(String user, IUserSocketContext ctx);
	boolean addListener(IServerSocketListener listener);
	boolean removeListener(IServerSocketListener listener);
}

File: csharp/CSharpLTS/Common/Transport/IServerUserSocketService.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using Common.Basic;

namespace Common.Transport
{
    public interface IServerUserSocketService : IPlugin
    {
        IUserSocketContext GetContext(string key);
        List<IUserSocketContext> GetContextByUser(string user);
        void SetUserContext(string user, IUserSocketContext ctx);
        bool AddListener(IServerSocketListener listener);
        bool RemoveListener(IServerSocketListener listener);
    }
}
